src/Npgsql/NpgsqlRawCopyStream.cs

```diff
@@ -89,49 +89,100 @@ internal NpgsqlRawCopyStream(NpgsqlConnector connector, string copyCommand)
 
         #region Write
 
-        public override void Write(byte[] buffer, int offset, int count) => Write(buffer, offset, count, false).GetAwaiter().GetResult();
+        public override void Write(byte[] buffer, int offset, int count)
+            => Write(new ReadOnlySpan<byte>(buffer, offset, count));
 
         public override Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
-        {
-            if (cancellationToken.IsCancellationRequested)
-                return Task.FromCanceled(cancellationToken);
-            using (NoSynchronizationContextScope.Enter())
-                return Write(buffer, offset, count, true);
-        }
+            => WriteAsync(new Memory<byte>(buffer, offset, count), cancellationToken).AsTask();
```

Comment:
The IsCancellationRequested check and NoSynchronizationContextScope have disappeared - this have to be on all public async APIs (below as well).

Reply:
They are back now.
But for me it is now redundant with what is done in WriteAsync(Memory)...